lisa/util/__init__.py

```diff
@@ -657,3 +659,47 @@ def strip_strs(obj: Any, fields: List[str]) -> Any:
             value = value.strip() if isinstance(value, str) else value
             setattr(obj, field, value)
     return obj
+
+
+def check_get_attr(actual_object: Any, attr_name: str = "") -> Any:
+    if hasattr(actual_object, attr_name):
+        return getattr(actual_object, attr_name)
+    raise LisaException(f"attr {attr_name} not existing in object {actual_object}")
+
+
+def is_value_expected(
+    actual_object: Any,
+    expected: Any,
+    func_name: str,
+    attr_name: str = "",
+    test_kwargs: Optional[Dict[str, Any]] = None,
+) -> bool:
+    actual_value: Any
+    if test_kwargs is None:
+        test_kwargs = {}
+    func = check_get_attr(actual_object, func_name)
+    if attr_name:
+        attr_name_list = attr_name.split(".")
+        actual_value = check_get_attr(func(**test_kwargs), attr_name_list[0])
+        current = actual_value
+        for remain_attr_name in attr_name_list[1:]:
+            actual_value = check_get_attr(current, remain_attr_name)
+            current = actual_value
+    else:
+        actual_value = func(**test_kwargs)
+    if actual_value == expected:
+        return True
+    else:
+        return False
+
+
+def retry_till_timeout(func: Any, *func_args: Any, **kwargs: Any) -> None:
```

Comment:
Try this example. if the logic is complicated, which is not able to be handled by lamda in one line, use a function.

```python
from functools import partial
from time import sleep
from typing import Callable

from lisa import LisaException
from lisa.util.logger import get_logger
from lisa.util.perf_timer import create_timer

log = get_logger()


def check_till_timeout(
    func: Callable[..., bool], timeout_message: str, timeout: int = 20
) -> None:
    timer = create_timer()
    while timer.elapsed(False) < timeout:
        if func():
            break
        sleep(1)
    if timer.elapsed() >= timeout:
        raise LisaException(f"timeout: {timeout_message}")


def method_with_arguments(wait_for: int = 5) -> bool:
    log.info("method_with_arguments")
    if global_wait_timer.elapsed(False) < wait_for:
        return False
    else:
        return True


global_wait_timer = create_timer()
log.info("calling check_till_timeout example 1...")
check_till_timeout(
    partial(method_with_arguments, wait_for=2),
    timeout_message="check_till_timeout example 1 is timeout",
)

global_wait_timer = create_timer()
log.info("calling check_till_timeout example 2...")
try:
    check_till_timeout(
        partial(method_with_arguments),
        timeout_message="check_till_timeout example 2 is timeout",
        timeout=2,
    )
except LisaException as identifier:
    log.info(identifier)

global_wait_timer = create_timer()
log.info("calling lambda 1...")
check_till_timeout(
    lambda: False if global_wait_timer.elapsed(False) < 2 else True,
    timeout_message="lambda example is timeout",
)

global_wait_timer = create_timer()
log.info("calling lambda 2...")

try:
    check_till_timeout(
        lambda: False if global_wait_timer.elapsed(False) < 5 else True,
        timeout_message="lambda example is timeout",
        timeout=2,
    )
except LisaException as identifier:
    log.info(identifier)

```